include/opencensus.hrl

```diff
@@ -43,45 +43,45 @@
 
 -record(span, {
           %% name of the span
-          name                                    :: unicode:unicode_binary(),
+          name                                    :: unicode:unicode_binary() | '_',
 
           %% 128 bit int trace id
-          trace_id                                :: opencensus:trace_id() | undefined,
+          trace_id                                :: opencensus:trace_id() | undefined | '_',
 
           %% 64 bit int span id
-          span_id                                 :: opencensus:span_id() | undefined,
+          span_id                                 :: opencensus:span_id() | undefined | '_',
           %% 64 bit int parent span
-          parent_span_id                          :: opencensus:span_id() | undefined,
+          parent_span_id                          :: opencensus:span_id() | undefined | '_',
 
           %% 8-bit integer, lowest bit is if it is sampled
-          trace_options = 1                       :: integer() | undefined,
+          trace_options = 1                       :: integer() | undefined | '_',
 
-          kind = ?SPAN_KIND_UNSPECIFIED           :: opencensus:span_kind(),
+          kind = ?SPAN_KIND_UNSPECIFIED           :: opencensus:span_kind() | '_',
 
-          start_time                              :: wts:timestamp(),
-          end_time                                :: wts:timestamp() | undefined,
+          start_time                              :: wts:timestamp() | {'$1', '_'},
+          end_time                                :: wts:timestamp() | undefined | '_',
 
-          attributes = #{}                        :: opencensus:attributes(),
+          attributes = #{}                        :: opencensus:attributes() | '_',
 
           %% optional stacktrace from where the span was started
-          stack_trace                             :: opencensus:stack_trace() | undefined,
+          stack_trace                             :: opencensus:stack_trace() | undefined | '_',
```

Comment:
It is just for the matchspec defined in `oc_span_sweeper`. dialyzer doesn't understand when a record is being constructed with `'_'` used like wildcards for fields, so sadly it will warn unless you define them in the typespec.

Reply:
Wouldn't it be better to disable Dialyzer for that line instead? Otherwise it will look strange in docs.

Reply:
It worked out. Just had to ignore the 3 private functions in the module to get it working.